javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -941,6 +957,18 @@ protected void writeStreamStyleSerialization(JavaClass classBlock, ClientModel m
         // No-op, meant for StreamSerializationModelTemplate.
     }
 
+    protected void addGeneratedImport(Set<String> imports) {
+        if (JavaSettings.getInstance().isDataPlaneClient()) {
+            imports.add(Generated.class.getName());
+        }
+    }
+
+    protected void addGeneratedAnnotation(JavaContext classBlock) {
+        if (JavaSettings.getInstance().isDataPlaneClient()) {
+            classBlock.annotation("Generated");
```

Comment:
The string "Generated" is used in several places. We should have a constant instead.